pkg/plugin/local/local.go

```diff
@@ -70,6 +81,13 @@ func (lds *LocalDataSource) updateDataFile() error {
 }
 
 func (lds *LocalDataSource) AddTrustZone(trustZone *trust_zone_proto.TrustZone) (*trust_zone_proto.TrustZone, error) {
+	if trustZone.Id == nil || *trustZone.Id == "" {
```

Comment:
I'd suggest failing if the resource already has an ID - it's the data source's responsibility to generate one.

Reply:
good plan!